client/object_hash.go

```diff
@@ -26,6 +26,8 @@ type PrmObjectHash struct {
 	csAlgo v2refs.ChecksumType
 
 	signer neofscrypto.Signer
+
+	isSessionIgnored bool
```

Comment:
Maybe we can move these session things into an internal struct of its own included in appropriate external structures? This would automatically unify the implementation and deduplicate code.

Reply:
The real session data is located in `meta v2session.RequestMetaHeader`. This header is a part of `api-go`. In this case, we should move `meta` and `isSessionIgnored` together to achieve unification. It will affect some other files, is it ok?

Reply:
Moved to a separate structure. There were not so many changes

Reply:
> +118 −152

That's what I call a good PR.